File: content-solutions/bfs/news-order.js
function solution(n, pairs, start) {
  const graph = Array.from({ length: n }, () => []);
  for (const [a, b] of pairs) {
    graph[a].push(b);
    graph[b].push(a);
  }
  for (const neighbors of graph) neighbors.sort((p, q) => p - q);
  const visited = new Array(n).fill(false);
  visited[start] = true;
  const queue = [start];
  let head = 0;
  while (head < queue.length) {
    const v = queue[head++];
    for (const w of graph[v]) {
      if (!visited[w]) {
        visited[w] = true;
        queue.push(w);
      }
    }
  }
  return queue;
}

File: content-solutions/bfs/news-order.py
from collections import deque


def solution(n, pairs, start):
    graph = [[] for _ in range(n)]
    for a, b in pairs:
        graph[a].append(b)
        graph[b].append(a)
    for neighbors in graph:
        neighbors.sort()
    visited = [False] * n
    visited[start] = True
    queue = deque([start])
    order = []
    while queue:
        v = queue.popleft()
        order.append(v)
        for w in graph[v]:
            if not visited[w]:
                visited[w] = True
                queue.append(w)
    return order
